modules/preview-react/pill/lib/Pill.tsx

```diff
@@ -0,0 +1,208 @@
+    },
+    disabled: {},
+  };
+};
+
+const pillBaseStyles: CSSObject = {
+  display: 'inline-flex',
+  alignItems: 'center',
+  borderRadius: borderRadius.m,
+  flexShrink: 0,
+  ...type.levels.subtext.large,
+  boxShadow: 'none',
+  outline: 'none',
+  fontWeight: 500,
+  WebkitFontSmoothing: 'antialiased',
+  MozOsxFontSmoothing: 'grayscale',
+  width: 'fit-content',
+  padding: `2px ${space.xxs}`,
+  height: space.m,
+  position: 'relative',
+};
+
+const StyledBasePill = styled(BaseButton.as('button'))<StyledType & PillProps>(
+  {
+    ...pillBaseStyles,
+
+    '&:active, &active:hover': {
+      'span[data-count="ck-pill-count"]': {
+        backgroundColor: colors.soap600,
+      },
+    },
+  },
+  boxStyleFn
+);
+
+const StyledNonInteractivePill = styled(StyledBasePill)({
+  cursor: 'default',
+  overflow: 'revert', // override BaseButton overflow styles so the click target exists outside the pill for removable
+});
+
+export const Pill = createComponent('button')({
+  displayName: 'Pill',
+  Component: ({children, model, ...config}: PillProps, ref, Element) => {
+    const value = useDefaultModel(model, config, usePillModel);
+    const {onClick, onDelete, maxWidth, ...elemProps} = config;
+    const actualEl = useConstant(() => {
+      if (Element === 'button') {
+        return !!onDelete || !onClick ? 'span' : 'button';
+      }
+      return Element;
+    });
+    return (
+      <PillModelContext.Provider value={value}>
+        <>
+          {!config.onClick && !config.onDelete && (
+            <StyledNonInteractivePill
+              maxWidth={value.state.maxWidth}
+              as={actualEl}
+              ref={ref}
+              border={`1px solid ${colors.licorice200}`}
+              {...elemProps}
+            >
+              <PillLabel>{children}</PillLabel>
+            </StyledNonInteractivePill>
+          )}
+          {config.onClick && !config.onDelete && (
+            <StyledBasePill
+              colors={getButtonPillColors()}
+              as={Element}
+              ref={ref}
+              onClick={config.onClick}
+              {...elemProps}
+            >
+              <HStack spacing="xxxs">
+                {React.Children.map(children, (child, index) => {
+                  if (typeof child === 'string') {
+                    return <PillLabel key={index}>{child}</PillLabel>;
+                  }
+                  return child;
+                })}
+              </HStack>
+            </StyledBasePill>
+          )}
+          {config.onDelete && !config.onClick && (
+            <StyledNonInteractivePill
+              colors={getRemovablePillColors(config.disabled)}
+              as={actualEl}
+              ref={ref}
+              backgroundColor={config.disabled ? colors.soap100 : colors.soap300}
+              color={config.disabled ? colors.licorice100 : colors.blackPepper400}
+              borderColor={config.disabled ? colors.licorice100 : colors.licorice200}
```

Comment:
Because it's a span element, the `colors` prop doesn't quite cover all the bases and I had to do some overrides for colors